RecoParticleFlow/PFClusterProducer/interface/PFHGCalRecHitCreator.h

```diff
@@ -58,8 +59,15 @@ template <typename DET,PFLayer::Layer Layer,ForwardSubdetector subdet>
 	double energy = hgrh.energy();
 	double time = hgrh.time();	
 	
-	const FlatTrd *thisCell = 
-	  static_cast<const FlatTrd*>(hgcGeo.getGeometry(detid));	
+	const CaloCellGeometry *thisCell = nullptr;
+	
+	if( DetId::Forward == detid.det() ) {
+	  const auto* hgGeom = static_cast<const HGCalGeometry*>(geom->getSubdetectorGeometry(detid.det(),detid.subdetId()));
+	  thisCell = static_cast<const FlatTrd*>(hgGeom->getGeometry(detid));
```

Comment:
The response is not quite in the direction I was asking about.
It was really just a c++ question about apparent redundancy in the implementation.

it looks like the full if-else here is equivalent to 
``` thisCell = geom->getSubdetectorGeometry(detid.det(),detid.subdetId())->getGeometry(detid);```
static_cast will do nothing in both the geometry cast and the cell cast.